lib/optimizely/decision_service.rb

```diff
@@ -417,6 +417,28 @@ def get_forced_variation(project_config, experiment_key, user_id)
       [variation, decide_reasons]
     end
 
+    def validated_forced_decision(context, user_context)
+      decision = user_context.find_forced_decision(context)
+      flag_key = context[:flag_key]
+      rule_key = context[:rule_key]
+      variation_key = decision ? decision[:variation_key] : decision
+      reasons = []
+      target = rule_key ? "flag (#{flag_key}), rule (#{rule_key})" : "flag (#{flag_key})"
+      if variation_key
+        variation = user_context.optimizely_client.get_flag_variation(flag_key, variation_key, 'key')
+        if variation
+          reason = "Variation (#{variation_key}) is mapped to #{target} and user (#{user_context.user_id}) in the forced decision map."
+          reasons.push(reason)
+          return variation, reasons
```

Comment:
this code can be moved to line 436, that way we won't have `reasons.push(reason)` twice.